tests/testthat/test-rd-html.R

```diff
@@ -346,3 +346,28 @@ test_that("titles don't get autolinked code", {
   rd <- rd_text("\\title{\\code{foo()}}", fragment = FALSE)
   expect_equal(extract_title(rd), "<code>foo()</code>")
 })
+
+# Handling URL failures verbosely -----------------------------------------
+
+
+test_that("URLs handled well", {
+
+  ## Binding the rdname to the context environment allows
+  # stop_on_bad_html_tag() to grab the filename via rlang
+  rlang::env_bind(.env = context, rdname = "test-rd-html.R")
```

Comment:
Re #771, @hadley could you please verify that this call to `env_bind()` won't have any undesirable side effects? I added it to ensure that the tests throw the correct error message, but the line can be omittted if you wish. Thank you!

Reply:
You can use `scoped_file_context()` which was designed for this exact scenario 😄